include/aspect/particle/output/hdf5.h

```diff
@@ -104,12 +102,13 @@ namespace aspect
           void
           load (std::istringstream &is);
 
-        private:
           /**
-           * Internal index of file output number.
+           * Returns a constructed filename.
            */
-          unsigned int file_index;
+          const std::string
+          get_file_name ();
```

Comment:
Also, since the function is virtual (because it is virtual in the base class), please mark it as virtual.